superdesk/storage/amazon/amazon_media_storage.py

```diff
@@ -149,17 +150,17 @@ def put(self, content, filename=None, content_type=None, resource=None, metadata
         of the stored file will be returned. The content type argument is used
         to appropriately identify the file when it is retrieved.
         """
-        filename = self.name_for_media(filename)
-        logger.debug('Going to save media file with %s ' % filename)
-        found = self._check_exists(filename)
+        logger.debug('Going to save file file=%s media=%s ' % (filename, _id))
+        _id = _id or self.media_id(filename)
+        found = self._check_exists(_id)
         if found:
-            return filename
+            return _id
 
         try:
             file_metadata = self.transform_metadata_to_amazon_format(metadata)
-            self.client.put_object(Key=filename, Body=content, Bucket=self.container_name,
+            self.client.put_object(Key=_id, Body=content, Bucket=self.container_name,
```

Comment:
Why are these keyword parameter capitalized? Is `boto3.client` an external library that defines them like this?